nexus/db-queries/src/db/queries/region_allocation.rs

```diff
@@ -316,7 +316,11 @@ impl CandidateZpools {
         // into a BigInt, not a Numeric. I welcome a better solution.
         let it_will_fit = (old_zpool_usage::dsl::size_used
             + diesel::dsl::sql(&zpool_size_delta.to_string()))
-        .le(diesel::dsl::sql(zpool_dsl::total_size::NAME));
+        .le(diesel::dsl::sql(
```

Comment:
Hours spent trying to make this work with Diesel, getting pushed away from subqueries, fighting aliases, and then fumbling with full-table scans as I tried to split this into a separate query:

~4

I am still motivated to make https://github.com/oxidecomputer/omicron/pull/5063 work after getting through this disk expungement stuff.